### PR TITLE
Use UTF-8 without BOM for SyntaxPrinter

### LINKED ISSUE #4912: `?sdl` returns file encoded as UTF-8 with BOM
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

I was trying out the new relay compiler and ran into a weird unsupported character issue. After some research I found out it is due to our `schema.graphql` file having `UTF-8 with BOM` and not just `UTF-8` encoding: https://github.com/facebook/relay/issues/3798#issuecomment-1085991645

### Steps to reproduce

Download your GraphQL schema like this: `curl localhost:5000/graphql?sdl -o ./src/schema.graphql`

### Relevant log output

_No response_

### Additional Context?

The issue is this: https://github.com/ChilliCream/hotchocolate/blob/61936613166e64665ab29d873c965a68084eb0df/src/HotChocolate/Language/src/Language.SyntaxTree/Utilities/SyntaxPrinter.cs#L68

It should be `new UTF8Encoding(false)` instead of `Encoding.UTF8`.

### Product

Hot Chocolate

### Version

13.0.0-preview.15

### PR DESCRIPTION
**Pull request by tobias-tengler**
- Writes schema to output stream using UTF-8 without BOM encoding
- Fixes snapshots impacted by this change
- Also removes some unused snapshot files

Closes #4912

### CODE REVIEW

**Comment by michaelstaib** [comment]
@tobias-tengler do not worry about the tests, we are on it .... we should be green again by next week.

**Comment by tobias-tengler** [comment]
Resolved all of the test failures that were caused by this change. So this should be good to go! :)

**Comment by tobias-tengler** [comment]
I think this would still be valuable.

File: src/HotChocolate/Language/src/Language.SyntaxTree/Utilities/SyntaxPrinter.cs [hunk 1]
```diff
@@ -65,9 +65,9 @@ public static string Print(this ISyntaxNode node, bool indented = true)
         CancellationToken cancellationToken = default)
     {
 #if NETSTANDARD2_0
-        using var streamWriter = new StreamWriter(stream, Encoding.UTF8);
+        using var streamWriter = new StreamWriter(stream, new UTF8Encoding(false));
```
Comment (glen-84) [inline src/HotChocolate/Language/src/Language.SyntaxTree/Utilities/SyntaxPrinter.cs:68]: The documentation recommends that `throwOnInvalidBytes` is set to `true`:

https://learn.microsoft.com/en-us/dotnet/api/system.text.utf8encoding?view=net-8.0

> To enable error detection and to make the class instance more secure, you should call the [UTF8Encoding(Boolean, Boolean)](https://learn.microsoft.com/en-us/dotnet/api/system.text.utf8encoding.-ctor?view=net-8.0#system-text-utf8encoding-ctor(system-boolean-system-boolean)) constructor and set the throwOnInvalidBytes parameter to true. With error detection enabled, a method that detects an invalid sequence of characters or bytes throws an [ArgumentException](https://learn.microsoft.com/en-us/dotnet/api/system.argumentexception?view=net-8.0) exception. Without error detection, no exception is thrown, and the invalid sequence is generally ignored.

https://learn.microsoft.com/en-us/dotnet/api/system.text.utf8encoding.-ctor?view=net-8.0

> For security reasons, you should enable error detection by calling a constructor that includes a throwOnInvalidBytes parameter and setting its value to true.

I've done that for #6751.

**Review by glen-84 — APPROVED** [review]

File: src/StrawberryShake/CodeGeneration/test/CodeGeneration.CSharp.Tests/Integration/StarWarsOnReviewSubCompletionTest.Client.cs [hunk 2]
```diff
@@ -445,7 +445,7 @@ public partial interface IStarWarsOnReviewSubCompletionClient
     public enum StarWarsOnReviewSubCompletionClientProfileKind
     {
         InMemory,
-        Default,
+        Default
```
Comment (michaelstaib) [inline src/StrawberryShake/CodeGeneration/test/CodeGeneration.CSharp.Tests/Integration/StarWarsOnReviewSubCompletionTest.Client.cs:448]:
```suggestion
        Default,
```

File: src/StrawberryShake/CodeGeneration/test/CodeGeneration.CSharp.Tests/Integration/StarWarsIntrospectionTest.Client.cs [hunk 3]
```diff
@@ -2834,7 +2834,7 @@ public enum __TypeKind
         /// <summary>
         /// Indicates this type is a non-null. `ofType` is a valid field.
         /// </summary>
-        NonNull,
+        NonNull
```
Comment (michaelstaib) [inline src/StrawberryShake/CodeGeneration/test/CodeGeneration.CSharp.Tests/Integration/StarWarsIntrospectionTest.Client.cs:2837]:
```suggestion
        NonNull,
```

File: src/StrawberryShake/CodeGeneration/test/CodeGeneration.CSharp.Tests/Integration/MultiProfileTest.Client.cs [hunk 4]
```diff
@@ -1149,7 +1149,7 @@ public enum Episode
     {
         NewHope,
         Empire,
-        Jedi,
+        Jedi
```
Comment (michaelstaib) [inline src/StrawberryShake/CodeGeneration/test/CodeGeneration.CSharp.Tests/Integration/MultiProfileTest.Client.cs:1152]:
```suggestion
        Jedi,
```

File: src/StrawberryShake/CodeGeneration/test/CodeGeneration.CSharp.Tests/Integration/MultiProfileTest.Client.cs [hunk 5]
```diff
@@ -1604,7 +1604,7 @@ public partial interface IMultiProfileClient
     public enum MultiProfileClientProfileKind
     {
         InMemory,
-        Default,
+        Default
```
Comment (michaelstaib) [inline src/StrawberryShake/CodeGeneration/test/CodeGeneration.CSharp.Tests/Integration/MultiProfileTest.Client.cs:1607]:
```suggestion
        Default,
```

**Review by michaelstaib — APPROVED** [review]

**Pull request merged**